routing-daemon/lib/openshift/routing/models/f5-icontrol-rest.rb

```diff
@@ -235,11 +185,80 @@ def delete_pool_alias pool_name, alias_str
-    end 
+      # The LTM must be configured with a default client SSL profile for fallback for custom alias SSL to work with SNI
+      # https://support.f5.com/kb/en-us/solutions/public/13000/400/sol13452.html
+
+      # write temp cert and key
+      certfname = "#{@tmpdir}/#{URI.escape(alias_str)}.crt"
+      keyfname = "#{@tmpdir}/#{URI.escape(alias_str)}.key"
+
+      File.write(certfname, ssl_cert)
+      File.write(keyfname, private_key)
+      begin
+        # scp cert and to F5 LTM (requires ssh key to be in authorized_keys on the F5 LTM
+        @logger.debug("Copying certificate for alias #{alias_str} for pool #{pool_name} to LTM host")
+        result = `scp -i #{@ssh_private_key} #{certfname} admin@#{@host}:/var/tmp/`
+        result = `scp -i #{@ssh_private_key} #{keyfname} admin@#{@host}:/var/tmp/`
+
+        @logger.debug("LTM cert to be installed /var/tmp/#{URI.escape(alias_str)}.crt")
+        post(url: "https://#{@host}/mgmt/tm/sys/crypto/cert",
+               payload: {
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-cert",
+                 "from-local-file" => "/var/tmp/#{URI.escape(alias_str)}.crt"
+               }.to_json)
+
+        @logger.debug("LTM cert to be installed /var/tmp/#{URI.escape(alias_str)}.key")
+        post(url: "https://#{@host}/mgmt/tm/sys/crypto/key",
+               payload: {
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-key",
+                 "from-local-file" => "/var/tmp/#{URI.escape(alias_str)}.key"
+               }.to_json)
+
+        @logger.debug("LTM creating client-ssl profile with #{URI.escape(alias_str)}.crt and #{URI.escape(alias_str)}.key")
+        post(url: "https://#{@host}/mgmt/tm/ltm/profile/client-ssl",
+                 payload: {
+                   "name" => "#{URI.escape(alias_str)}-ssl-profile",
+                   "cert" => "#{URI.escape(alias_str)}-https-cert.crt",
+                   "key" => "#{URI.escape(alias_str)}-https-key.key",
+                   "serverName" => "#{URI.escape(alias_str)}"
+                 }.to_json)
+
+        @logger.debug("LTM adding #{URI.escape(alias_str)}-ssl-profile client-ssl to #{@https_vserver}")
+        post(url: "https://#{@host}/mgmt/tm/ltm/virtual/#{@https_vserver}/profiles",
+                 payload: {
+                   "name" => "#{URI.escape(alias_str)}-ssl-profile",
+                   "context" => "clientside",
+                 }.to_json)
+
+        # Requires LTM System->Users->admin terminal setting to be set to advanced (bash)
+        @logger.debug("LTM removing temporary. rm -f /var/tmp/#{URI.escape(alias_str)}.crt")
+        result = `ssh -i #{@ssh_private_key} admin@#{@host} 'rm -f /var/tmp/#{URI.escape(alias_str)}.crt'`
+        @logger.debug("LTM removing temporary. rm -f var/tmp/#{URI.escape(alias_str)}.key")
+        result = `ssh -i #{@ssh_private_key} admin@#{@host} 'rm -f /var/tmp/#{URI.escape(alias_str)}.key'`
```

Comment:
Why not just use `/var/tmp/#{certfname}` and `/var/tmp/#{keyfname}` on these four lines?